lightning/src/ln/channelmanager.rs

```diff
@@ -4291,12 +4290,25 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 						} else { None };
 
 						let mut pending_events = self.pending_events.lock().unwrap();
+
+						let channel_id = prev_outpoint.to_channel_id(); 
+						let channel = match channel_state_lock.by_id.entry(channel_id) {
+							hash_map::Entry::Occupied(chan) => chan,
+							hash_map::Entry::Vacant(_) => {
+								panic!("Channel doesn't exist")
+							}
+						};
+						let source_node_id = channel.get().get_counterparty_node_id();
```

Comment:
@TheBlueMatt Getting the correct `source_node_id` and `channel_id` now. Only case remaining is when the channel has been closed. How do we get the `source_node_id` in that case? And should we still emit the id of the closed channel?

Reply:
Hmmmm, maybe lets just drop the source node_id entirely for now? Its kinda annoying to do a channel lookup just to get it, especially given we can't find it if we went onchain. Users can do the lookup manually, if they want.